src/algokit_utils/_transfer.py

```diff
@@ -67,18 +85,50 @@ def transfer(client: "AlgodClient", parameters: TransferParameters) -> PaymentTx
         note=parameters.note.encode("utf-8") if isinstance(parameters.note, str) else parameters.note,
         sp=suggested_params,
     )  # type: ignore[no-untyped-call]
+
+    result = _send_transaction(client=client, transaction=transaction, parameters=parameters)
+    assert isinstance(result, PaymentTxn)
+    return result
+
+
+def transfer_asset(client: "AlgodClient", parameters: TransferAssetParameters) -> AssetTransferTxn:
+    sender = address_from_private_key(parameters.from_account.private_key)  # type: ignore[no-untyped-call]
+    suggested_params = parameters.suggested_params or client.suggested_params()
+    xfer_txn = AssetTransferTxn(
+        sp=suggested_params,
+        sender=sender,
+        receiver=parameters.to_address,
+        close_assets_to=None,
+        revocation_target=parameters.clawback_from,
+        amt=parameters.amount,
+        note=parameters.note,
+        index=parameters.asset_id,
+        rekey_to=None,
+    )  # type: ignore[no-untyped-call]
+
+    result = _send_transaction(client=client, transaction=xfer_txn, parameters=parameters)
+    assert isinstance(result, AssetTransferTxn)
+    return result
+
+
+def _send_transaction(
+    client: "AlgodClient",
+    transaction: PaymentTxn | AssetTransferTxn,
+    parameters: TransferAssetParameters | TransferParameters,
+) -> PaymentTxn | AssetTransferTxn:
     if parameters.fee_micro_algos:
         transaction.fee = parameters.fee_micro_algos
 
-    if not suggested_params.flat_fee:
+    if parameters.suggested_params is not None and not parameters.suggested_params.flat_fee:
         _check_fee(transaction, parameters.max_fee_micro_algos)
```

Comment:
@inaie-makerx check_fee doesn't do anything with suggested_params whats the reason for adding an extra check here? This is also what's causing the test_transfer_algo_max_fee_fails test to fail